editor/src/components/canvas/controls/select-mode/post-action-menu.tsx

```diff
@@ -10,191 +21,403 @@ import { useDispatch } from '../../../editor/store/dispatch-context'
+        setOpen(false)
+      }
+    }, [postActionSessionChoices])
+
+    const dispatch = useDispatch()
+
+    React.useEffect(() => {
+      function handleKeyDown(event: KeyboardEvent) {
+        const isDismissKey = event.key === 'Enter' || event.key === 'Escape'
+
+        if (isDismissKey && isPostActionMenuActive(postActionSessionChoices)) {
+          event.preventDefault()
+          event.stopPropagation()
+          event.stopImmediatePropagation()
+
+          dispatch([clearPostActionData()])
+        }
+      }
+
+      if (isPostActionMenuActive(postActionSessionChoices)) {
+        window.addEventListener('keydown', handleKeyDown, true)
+      }
+
+      return function cleanup() {
+        window.removeEventListener('keydown', handleKeyDown, true)
+      }
+    }, [dispatch, postActionSessionChoices])
+
+    if (!isPostActionMenuActive(postActionSessionChoices)) {
+      return null
+    }
+
+    return (
+      <CanvasOffsetWrapper>
+        {when(
+          errorOverlayShown,
+          <div
+            style={{
+              backgroundColor: `rgba(255, 69, 2, 0.5)`,
+              position: 'absolute',
+              left: positioningProps.x - 34 / scale,
+              top: positioningProps.y - 12 / scale,
+              width: 30 / scale,
+              height: 30 / scale,
+              padding: 6 / scale,
+              display: 'flex',
+              borderRadius: 15 / scale,
+            }}
+          >
+            <div
+              style={{
+                backgroundColor: `rgba(255, 69, 2)`,
+                height: '100%',
+                contain: 'layout',
+                flexGrow: 1,
+                borderRadius: 9 / scale,
+              }}
+            />
+          </div>,
+        )}
+        <div
+          style={{
+            display: 'block',
+            pointerEvents: 'initial',
+            position: 'absolute',
+            fontSize: 9,
+            top: positioningProps.y,
+            left: positioningProps.x,
+          }}
+          onMouseDown={stopPropagation}
+          onClick={openIfClosed}
+        >
+          <FlexColumn
+            style={{
+              zoom: 1 / scale,
+              display: 'flex',
+              alignItems: 'stretch',
+              padding: 4,
+              borderRadius: 4,
+              background: colorTheme.bg0.value,
+              boxShadow: UtopiaStyles.popup.boxShadow,
+              cursor: open ? undefined : 'pointer',
+              fontSize: 10,
+            }}
+          >
+            {open ? (
+              <PostActionMenu postActionSessionChoices={postActionSessionChoices} />
+            ) : (
+              <Icn category='semantic' type='clipboard' color={'main'} width={18} height={18} />
+            )}
+          </FlexColumn>
+        </div>
+      </CanvasOffsetWrapper>
+    )
+  },
+)
+FloatingPostActionMenu.displayName = 'FloatingPostActionMenu'
+
+export const InspectorPostActionMenu = React.memo(() => {
```

Comment:
is this still used somewhere?

Reply:
great catch

Reply:
https://github.com/concrete-utopia/utopia/pull/3895/commits/858e248bf03b47ea33e6ca2cfb6f4982b76ec5d1